src/main/java/org/apache/sling/xss/impl/xml/Policy.java

```diff
@@ -0,0 +1,391 @@
+            logger.warn("Unsupported configuration directive {} is set to true and will be ignored", DIRECTIVE_EMBED_STYLE_SHEETS);
+        }
+        return antiSamyRules;
+    }
+
+    private void init(AntiSamyRules topLevelElement) throws PolicyException {
+        parseCommonRegExps(topLevelElement.getRegexpList());
+        parseDirectives(topLevelElement.getDirectiveList());
+        parseCommonAttributes(topLevelElement.getCommonAttributeList());
+        parseGlobalAttributes(topLevelElement.getGlobalTagAttributes().getGlobalTagAttributeList());
+        parseDynamicAttributes(topLevelElement.getDynamicTagAttribute().getDynamicTagAttributeList());
+        parseTagRules(topLevelElement.getTagRulesList());
+        parseCSSRules(topLevelElement.getPropertyList());
+
+        parseAllowedEmptyTags(topLevelElement.getAllowedEmptyTags());
+    }
+
+    /**
+     * Go through the <common-regexps> section of the policy file.
+     *
+     * @param root                      Top level of <common-regexps>
+     * @param commonRegularExpressions2 the antisamy regexp objects
+     */
+    private void parseCommonRegExps(List<Regexp> root) {
+        for (Regexp regex : root) {
+            String name = regex.getName();
+            Pattern regexp = Pattern.compile(regex.getValue(),
+                    Pattern.DOTALL);
+            commonRegularExpressions.put(name, regexp);
+        }
+    }
+
+    /**
+     * Go through <directives> section of the policy file.
+     *
+     * @param root       Top level of <directives>
+     * @param directives The directives map to update
+     */
+    private void parseDirectives(List<Directive> root) {
+        for (Directive directive : root) {
+            String name = directive.getName();
+            String value = directive.getValue();
+            directives.put(name, value);
+        }
+    }
+
+    private void parseCommonAttributes(List<Attribute> root) {
+
+        for (Attribute attribute : root) {
+
+            List<Regexp> allowedRegexps = getAllowedRegexps(attribute.getRegexpList());
+            Attribute newAttribute = new Attribute(attribute.getName(), allowedRegexps, attribute.getLiteralList(),
+                    attribute.getOnInvalid(), attribute.getDescription());
+            commonAttributes.put(attribute.getName().toLowerCase(), newAttribute);
+        }
+
+    }
+
+    // /**
+    // * Go through <allowed-empty-tags> section of the policy file.
+    // *
+    // * @param allowedEmptyTagsListNode Top level of <allowed-empty-tags>
+    // * @param allowedEmptyTags The tags that can be empty
+    // */
+    private void parseAllowedEmptyTags(AllowedEmptyTags allowedEmptyTagsList) throws PolicyException {
+        if (allowedEmptyTagsList != null) {
+            allowedEmptyTags = allowedEmptyTagsList.getLiterals();
+        } else
+            allowedEmptyTags.addAll(Arrays.asList(
```

Comment:
This does not look right, we should not hardcode a list of tags. Is it needed?

Reply:
As far as I understand antisamy has them as a default:
https://github.com/nahsra/antisamy/blob/f1e49b92b1d9f2506bd018b6f4fd4a0fac308d77/src/main/java/org/owasp/validator/html/scan/Constants.java

Reply:
Resolved: 81cc4a4f07ce77e8ac383fe3263a0787a2ffa447